runtime/controller/metrics.go

```diff
@@ -50,23 +51,36 @@ import (
 type Metrics struct {
 	Scheme          *runtime.Scheme
 	MetricsRecorder *metrics.Recorder
+	ownedFinalizers []string
 }
 
 // MustMakeMetrics creates a new Metrics with a new metrics.Recorder, and the Metrics.Scheme set to that of the given
-// mgr.
+// mgr, along with an optional set of owned finalizers which is used to determine when an object is being deleted.
 // It attempts to register the metrics collectors in the controller-runtime metrics registry, which panics upon the
 // first registration that causes an error. Which usually happens if you try to initialise a Metrics value twice for
 // your controller.
-func MustMakeMetrics(mgr ctrl.Manager) Metrics {
+func MustMakeMetrics(mgr ctrl.Manager, finalizers ...string) Metrics {
 	metricsRecorder := metrics.NewRecorder()
 	crtlmetrics.Registry.MustRegister(metricsRecorder.Collectors()...)
 
 	return Metrics{
 		Scheme:          mgr.GetScheme(),
 		MetricsRecorder: metricsRecorder,
+		ownedFinalizers: finalizers,
 	}
 }
 
+// IsDelete returns if the object is deleted by checking for deletion timestamp
+// and owned finalizers in the object.
+func (m Metrics) IsDelete(obj conditions.Getter) bool {
+	for _, f := range m.ownedFinalizers {
+		if slices.Contains(obj.GetFinalizers(), f) {
```

Comment:
Why pick this over https://github.com/kubernetes-sigs/controller-runtime/blob/5aa6a71fc79d4058b26355eb58bdf2430fce423e/pkg/controller/controllerutil/controllerutil.go#L384C5-L384C23?